internal/app/wwctl/overlay/edit/main.go

```diff
@@ -122,16 +115,14 @@ func CobraRunE(cmd *cobra.Command, args []string) error {
 		}
 	}
 
-	destination, destinationErr := os.OpenFile(overlayFile, os.O_RDWR|os.O_CREATE, os.FileMode(PermMode))
-	if destinationErr != nil {
-		wwlog.Error("Unable to update %s: %s", overlayFile, destinationErr)
+	err := os.Chmod(tempFile.Name(), os.FileMode(0644))
```

Comment:
I think this PR looks good in general (with my testing so far, anyway) but I don't think this chmod is necessary; when I used this PR with the chmod removed, I got a file with 0600, which I think is even better (and I don't like the 0644 being hardcoded).

```
-rw-------. 1 root root       0 Jul  2 16:58 testfile
-rw-------. 1 root root     669 Jul  2 16:59 testfile.ww
```

Let me know what you think, though.

Reply:
thanks for that! reflected in this commit
https://github.com/warewulf/warewulf/pull/1238/commits/b61dd17dbcec7d1c37d791318a7f1a14512b53ef